src/_guides/language/async-await.md

```diff
@@ -0,0 +1,344 @@
+* A future can be in one of two states: uncompleted, or completed. 
+* A completed future can either (1) complete with a result or (2) complete with an error
+* When a function that returns a future is invoked, two things happen:
+
+1. The function queues up work to be done and returns an uncompleted future.
+2. Later, when the operation is finished, the future completes with a value or with an error.
+
+
+## Working with Futures in Dart: async and await
+
+
+
+In the previous section you learned that, in Dart, future objects represent the results of asynchronous operations -- if a function does asynchronous work, it returns a future object. But how do we access the results of future objects when using asynchronous functions? 
+
+The `async` and `await` keywords provide a declarative way to (1) define asynchronous functions and (2) access their results  *after they have completed.* 
+
+
+To define an `async` function: 
+
+1. Add the `async` keyword after the function's parameters but before the function's body. 
+
+    createOrderMessage () async {
+
+2. Update the function's type signature to return a future. In the following example, we define an `async` function that completes with a string.   
+
+    Future<String> createOrderMessage () async {
+
+ *Note: You can *  [use Future<void>](https://dart.dev/guides/language/effective-dart/design#do-use-futurevoid-as-the-return-type-of-asynchronous-members-that-do-not-produce-values) * as the return type for asynchronous functions that don't return usable values. * 
+
+
+Once you have defined an async function, that function will return a future object. Using the await keyword, you can "wait" for the completed future to return either its value or its error.
+
+    String order = await getUserOrder();
+
+Remember, you can only use the `await` keyword within an `async` function body. The following example demonstrates how to convert `createOrderMessage` from synchronous to asynchronous:
+
+```
+// BEFORE: handle a function that returns a string
+
+String createOrderMessage () {
+  var order = getUserOrder();
+  return 'Your order is: $order';
+}
+
+main() {
+  print(createOrderMessage()); 
+}
+```
+
+```
+// After: handle an async function that returns a Future<String>
+
+Future<String> createOrderMessage () async {
+  var order = await getUserOrder();
+  return 'Your order is: $order';
+}
+
+main() async {
+  print(await createOrderMessage());
+}
+```
+
+*  *Note: *  [All functions return a value](https://dart.dev/guides/language/language-tour#return-values) * even if you don't explicitly use the *  *`return`*  * keyword. * 
+*  *Note: Dart can *  [infer the Future<String> type](https://dart.dev/guides/language/sound-dart#type-inference) * for you.
```

Comment:
I worry that `Dart infers the Future type for you` might mislead the reader into thinking that Dart *always* infers the Future type for you. I've linked to the rules regarding type inference so that we don't duplicate efforts across docs. 

For now I've left the `can` in since I think this is somewhat more true to the language. LMK if you'd like to discuss more.